kin-devplatform-sdk/src/main/java/kin/devplatform/settings/presenter/SettingsPresenter.java

```diff
@@ -177,16 +184,48 @@ private void switchAccount(int accountIndex) {
 		accountManager.switchAccount(accountIndex, new KinCallback<Boolean>() {
 			@Override
 			public void onResponse(Boolean response) {
+				view.stopWaiting();
 				eventLogger.send(RestoreWalletCompleted.create());
 			}
 
 			@Override
 			public void onFailure(KinEcosystemException exception) {
+				view.stopWaiting();
 				eventLogger.send(GeneralEcosystemSdkError
 					.create(ErrorUtil.getPrintableStackTrace(exception), String.valueOf(exception.getCode()),
 						"SettingsPresenter.switchAccount onFailure"));
 				showCouldNotImportAccountError();
 			}
+		}, new IMigrationManagerCallbacks() {
+
+			@Override
+			public void onMigrationStart() {
+				if (view != null) {
+					// TODO: 07/02/2019 confirm with Ayelet
+					view.showMigrationStartedDialog();
+				}
+			}
+
+			@Override
+			public void onReady(IKinClient kinClient) {
+				if (view != null) {
+					// TODO: 07/02/2019 Maybe if it is not started then show different message because there were no actual migration.
+					// TODO: 07/02/2019 Although it can start even if it was migrated, for example if the account was migrated but currently not saved locally in the device.
+					// TODO: 07/02/2019 It can be solved only if we will add a bit complicate method which checks if an account is already migrated.
+
+					// TODO: 07/02/2019 Also maybe it will be correct to update the user that he can go back and use the app only after the wallet address is finished and not here...
```

Comment:
yes, you can't tell him the account is ready, cause the process can fail on update wallet, and he will see an error (is it so?)

Reply:
Currently he will not see an error if wallet address update has failed.
I didn't add something as it was not requested.
Will talk about it with Ayelet and will see if she wants.
It is pretty easy to add an error dialog in that case, the question is what the user will do?
What we will let him to do? retry? maybe something else...